controllers/datadogagent/override/daemonset.go

```diff
@@ -16,6 +16,15 @@ func DaemonSet(daemonSet *v1.DaemonSet, override *v2alpha1.DatadogAgentComponent
 	if override.Name != nil && *override.Name != "" {
 		daemonSet.Name = *override.Name
 	}
+	if override.Strategy != nil {
+		if override.Strategy.Type == "RollingUpdate" {
```

Comment:
I don't think we need to restrict this to only be set if the strategy type is `RollingUpdate` since this would require the user to set `strategy.type` to `RollingUpdate` if they want to override `MaxUnavailable`/`MaxSurge`. Kubernetes will default this to `RollingUpdate` when not set, so I think we should allow a user to have the configuration:
```yaml
      strategy: 
        rollingUpdate:
          maxUnavailable: "2%"
```
and kubernetes will fill in the `type` in the daemonset:
```
  updateStrategy:
    rollingUpdate:
      maxSurge: 0
      maxUnavailable: 2%
    type: RollingUpdate
```

Even if type `OnDelete` is configured along with `rollingUpdate` configs, Kubernetes doesn't seem to throw any errors. I think the rollingUpdate is ignored for type `OnDelete` within the Kubernetes code so it shouldn't be an issue even if someone tries to set this kind of config:
```yaml
  updateStrategy:
    type: OnDelete
    rollingUpdate:
      maxUnavailable: 1
```